### PR TITLE
fix: remove dependency-tree flag for image subcommand

### LINKED ISSUE #2491: --dependency-tree option not working using trivy image
## Description

I'm unable to see the `--dependency-tree` option working using `trivy image --dependency-tree myimage`

## What did you expect to happen?
After the node libraries vulnerabilities, we should see a dependency tree starting as:
```
Dependency Origin Tree
======================
package-lock.json 
...
```

## What happened instead?
As it can be seen in the output, `trivy image` shows the table but not the tree.

## Output of run with `-debug`:

```
trivy --debug image --dependency-tree --vuln-type library --security-checks vuln --ignore-unfixed --severity HIGH,CRITICAL node:12.22.12-slim
2022-07-11T09:06:01.924+0200	DEBUG	'--dependency-tree' enables '--list-all-pkgs'.
2022-07-11T09:06:01.924+0200	DEBUG	Severities: HIGH,CRITICAL
2022-07-11T09:06:01.972+0200	DEBUG	cache dir:  /Users/rcarpio/Library/Caches/trivy
2022-07-11T09:06:01.972+0200	DEBUG	DB update was skipped because the local DB is the latest
2022-07-11T09:06:01.972+0200	DEBUG	DB Schema: 2, UpdatedAt: 2022-07-11 06:07:28.987601309 +0000 UTC, NextUpdate: 2022-07-11 12:07:28.987601109 +0000 UTC, DownloadedAt: 2022-07-11 07:05:26.972549 +0000 UTC
2022-07-11T09:06:01.973+0200	INFO	Vulnerability scanning is enabled
2022-07-11T09:06:01.973+0200	DEBUG	Vulnerability type:  [library]
2022-07-11T09:06:01.986+0200	DEBUG	Image ID: sha256:9b792ac6810e5c2cc2e28612129101c1ea2922cf322b708e8f08a93905a0729d
2022-07-11T09:06:01.986+0200	DEBUG	Diff IDs: [sha256:dc8ea258c7d2a42fbd972e96b5830c6618237dc9a1302d06ce6a1a9a3b972578 sha256:3d1236fa695c6ed62f42f051179e9d6916e25d2c903fcd0de3afc74c2ebe37aa sha256:56ff4c5413ca0a7e1372ee3e0bd3ae0d3acea665a316bf08faa4f4d711ae9666 sha256:7722feb318faa1518c0036c35e0d9916b1e81059acb76a357274d78d0fc3eb09 sha256:a2300a3fbe9961739214ae0c1a4baa8b3df47a50c0141c2d350c0cb41f85d4fa]
2022-07-11T09:06:01.986+0200	DEBUG	Base Layers: [sha256:dc8ea258c7d2a42fbd972e96b5830c6618237dc9a1302d06ce6a1a9a3b972578]
2022-07-11T09:06:01.993+0200	INFO	Number of language-specific files: 1
2022-07-11T09:06:01.993+0200	INFO	Detecting node-pkg vulnerabilities...
2022-07-11T09:06:01.993+0200	DEBUG	Detecting library vulnerabilities, type: node-pkg, path:
2022-07-11T09:06:01.999+0200	INFO	Table result includes only package filenames. Use '--format json' option to get the full path to the package file.

Node.js (node-pkg)

Total: 3 (HIGH: 2, CRITICAL: 1)

┌───────────────────────────┬────────────────┬──────────┬───────────────────┬────────────────────────────┬─────────────────────────────────────────────────────────┐
│          Library          │ Vulnerability  │ Severity │ Installed Version │       Fixed Version        │                          Title                          │
├───────────────────────────┼────────────────┼──────────┼───────────────────┼────────────────────────────┼─────────────────────────────────────────────────────────┤
│ ansi-regex (package.json) │ CVE-2021-3807  │ HIGH     │ 3.0.0             │ 3.0.1, 4.1.1, 5.0.1, 6.0.1 │ nodejs-ansi-regex: Regular expression denial of service │
│                           │                │          │                   │                            │ (ReDoS) matching ANSI escape codes                      │
│                           │                │          │                   │                            │ https://avd.aquasec.com/nvd/cve-2021-3807               │
│                           │                │          ├───────────────────┤                            │                                                         │
│                           │                │          │ 4.1.0             │                            │                                                         │
│                           │                │          │                   │                            │                                                         │
│                           │                │          │                   │                            │                                                         │
├───────────────────────────┼────────────────┼──────────┼───────────────────┼────────────────────────────┼─────────────────────────────────────────────────────────┤
│ minimist (package.json)   │ CVE-2021-44906 │ CRITICAL │ 1.2.5             │ 1.2.6                      │ minimist: prototype pollution                           │
│                           │                │          │                   │                            │ https://avd.aquasec.com/nvd/cve-2021-44906              │
└───────────────────────────┴────────────────┴──────────┴───────────────────┴────────────────────────────┴─────────────────────────────────────────────────────────┘
```

## Output of `trivy -v`:

```
trivy --version
Version: 0.29.2
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-07-08 12:06:33.858874338 +0000 UTC
  NextUpdate: 2022-07-08 18:06:33.858873938 +0000 UTC
  DownloadedAt: 2022-07-08 14:03:09.041507 +0000 UTC
```

## Additional details (base image name, container registry info...):
We talk about it in slack: https://aquasecurity.slack.com/archives/C02KVB6AED9/p1657028504009709

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
`--dependency-tree` flag is currently only supported by `fs` or `repo` commands. Also, only `package-lock.json` files are currently supported.
- Remove `--dependency-tree` flag for `image` and `sbom` subcommand.
- Update docs(add info about supported subcommands)
- Add info message when using `--dependency-tree` flag.

## Related issues
- Close #2491 

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by itaysk** [comment]
should/can we also warn if scanning an unsupported language ("--dependency-tree" can be used only with nodejs dependencies)

**Comment by DmitriyLewen** [comment]
This is good idea. I added this warning.

**Comment by knqyf263** [comment]
@DmitriyLewen The current implementation already shows the warning if it has result other than node.js.
@itaysk It's already mentioned in doc. I think updating flag description is enough.

**Comment by DmitriyLewen** [comment]
@knqyf263 we can show warning if report doesn't contain node.js file. What do you this about it?

**Comment by knqyf263** [comment]
I'm concerned a repository may mix some languages. Let's say a repository contains package-lock.json and Pipfile.lock. User wonder why a tree is shown for package-lock.json and not for Pipfile.lock because the warn message will not be displayed. I think user should know which ecosystem is supported anyway. We can show an info message every time `--dependency-tree` is specified.

File: pkg/report/writer.go [hunk 1]
```diff
@@ -51,6 +52,19 @@ func Write(report types.Report, option Option) error {
 	var writer Writer
 	switch option.Format {
 	case FormatTable:
+		if option.Tree {
+			switch report.ArtifactType {
+			case ftypes.ArtifactFilesystem, ftypes.ArtifactRemoteRepository: // only `fs` and `repo` commands support `--dependency-tree` flag
```
Comment (knqyf263) [inline pkg/report/writer.go:57]: After consideration, I found we should disable `--dependency-tree` in the `image` subcommand. You can put nil as below.
https://github.com/aquasecurity/trivy/blob/ed322ac6664306c13ed673e0fa7dd6f76ebc9841/pkg/commands/app.go#L531-L533

Comment (DmitriyLewen) [reply pkg/report/writer.go]: Thanks! i missed it. Now we don't need warning, because user will just get `unknown flag` error.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**